app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/CharacterServicesImpl.kt

```diff
@@ -8,6 +8,18 @@ import io.reactivex.Observable
 
 class CharacterServicesImpl(private val api: MarvelResquestGenerator = MarvelResquestGenerator(), private val mapper: CharacterMapperService = CharacterMapperService()) {
 
+    fun getCharacterById(id: Int?): Observable<Character> {
+        return Observable.create { subscriber ->
+            val callResponse = api.createService(MarvelApi::class.java).getCharacterById(id!!)
```

Comment:
do not use !!, please.. check if it is needed that the id is nullable or not. if it is needed to use an id.let{ ....} or something like that

Reply:
Done